subsys/bluetooth/controller/ll_sw/ull_conn_iso.c

```diff
@@ -112,6 +114,74 @@ struct ll_conn_iso_stream *ll_iso_stream_connected_get(uint16_t handle)
 	return cis;
 }
 
+struct ll_conn_iso_stream *ll_conn_iso_stream_get_by_acl(struct ll_conn *conn, uint16_t *cis_iter)
+{
+	uint8_t cis_iter_start = (cis_iter == NULL) || (*cis_iter) == UINT16_MAX;
+	uint8_t cig_hndl;
+
+	/* Find CIS associated with ACL conn */
+	for (cig_hndl = 0; cig_hndl < CONFIG_BT_CTLR_CONN_ISO_GROUPS; cig_hndl++) {
+		struct ll_conn_iso_stream *cis;
+		struct ll_conn_iso_group *cig;
+		uint16_t handle_iter;
+		int8_t cis_idx;
+
+		cig = ll_conn_iso_group_get(cig_hndl);
+		if (!cig) {
+			continue;
+		}
+
+		handle_iter = UINT16_MAX;
+
+		for (cis_idx = 0; cis_idx < cig->lll.num_cis; cis_idx++) {
+			cis = ll_conn_iso_stream_get_by_group(cig, &handle_iter);
+			LL_ASSERT(cis);
+
+			uint16_t cis_handle = cis->lll.handle;
+
+			cis = ll_iso_stream_connected_get(cis_handle);
+			if (!cis) {
+				continue;
+			}
+
+			if (!cis_iter_start) {
+				/* Look for iterator start handle */
+				cis_iter_start = cis_handle == (*cis_iter);
+			} else if (cis->lll.acl_handle == conn->lll.handle) {
```

Comment:
I am not sure why there is an "else" here. Can you maybe explain?

Reply:
The idea is that we can iterate through all streams of all groups, returning those with the "right" ACL handle. For that reason we can't just use an index. By use of the `cis_iter_start`, a state is maintained to say if we reached the start criterion for finding the next match. By using `cis_iter` as [in/out], we search for the previous match before "starting" search for the next. So until `cis_iter_start` is set, it just loops, after it is set, then we start comparing handles.

Reply:
> No objections, but a bit too many open questions from my side to approve.

I think the open questions have been addressed for now.

Reply:
On reading the code again, I think I got my head around this part now :-)